jest/async-storage-mock.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @format
+ * @flow
+ */
+
+type KeyValueType = Array<[string, any]>;
+
+type CallbackType = (error: ?Error) => void;
+
+type KeysType = Array<string>;
+
+type GetKeysType = (error: ?Error, keys: KeysType) => void;
+
+type MultiGetType = (error: ?Error, result: KeyValueType) => void;
+
+const AsyncStorageMock = {
+  /**
+   * Clears all keys in storage
+   */
+  clear: (cb: CallbackType) => cb(),
```

Comment:
So the mock is actually test-framework agnostic. Maybe it's nice, but I think most of the users would expect this mock to be fully functional in JS-land only, plus it would be nice if it used `jest.fn` to mock methods, because that would make them, well, mocks :P which you can inspect.

How about getting rid of the example mock and just provide the basic implementation + mocks here?

Reply:
That was my dilemma here - to provide fully functional mock or simple one + docs about providing your own functionality. I decided to do the latter, as I thought most people would use it to pass tests. 

But I think you might be right here - it's much easier for users to implement it other way around. Going to make it fully functional first.

Also, I had `jest.fn` at first, but then I thought about accessing this mock in your test - how would you do that? You'd have to import mock into your test and check, i.e if it was called with such arguments. Or am I missing something here?

Reply:
If you substitute module with a mock like `RN.AsyncStorage = ASMock;`, `AsyncStorage` should be a mock then and you'll be able to do `expect(AsyncStorage.getItem).toBeCalledWith(smth)`

Reply:
Yeah, but this is mock for `NativeModule`, not something that user has access too